### PR TITLE
Update filter page and API endpoints

### PR DESCRIPTION
**Pull request by dmitryduev**
In this PR:

- Updated Kowalski filter versioning API to address dmitryduev/kowalski#71
- Using [`penquins`](https://github.com/dmitryduev/penquins) library to query Kowalski
- Added tags to F-specific SP OpenAPI endpoint specs
- Re-skinned the Filter page; conditional rendering + avoiding showing errors if filter does not exist on Kowalski
- Added toggle switches for `update_annotations` and `autosave` options
- Copy filter def's to clipboard
- Mentioned the `PS1_STRM` in the docs

TODO:
- <s>merge dmitryduev/kowalski#77</s>
- tests!

filters-1

### CODE REVIEW

File: extensions/skyportal/skyportal/handlers/api/kowalski_filter.py [hunk 1]
```diff
@@ -269,41 +262,40 @@ def patch(self, filter_id):
...
 
-        if resp.status_code == requests.codes.ok:
-            data = bj.loads(resp.text).get("data")
+        if status == "success":
             return self.success(data=data)
-        else:
-            message = bj.loads(resp.text).get("message")
-            log(f"Failed to update filter on Kowalski: {message}")
-            return self.error(f"Failed to update filter on Kowalski: {message}")
+        return self.error(message=message)
```
Comment (guynir42) [inline extensions/skyportal/skyportal/handlers/api/kowalski_filter.py:289]: See previous comment...

File: extensions/skyportal/static/js/components/Filter.jsx [hunk 2]
```diff
@@ -211,196 +296,316 @@ const Filter = () => {
     );
   }
 
+  const highlightSyntax = str => (
+    <pre
+      style={{ display: 'inline', fontSize: "0.75rem", fontFamily: "Lucida Console, sans-serif" }}
+      dangerouslySetInnerHTML={{
+        __html: str
+        // __html: Prism.highlight(str, Prism.languages.javascript, "javascript"),
```
Comment (guynir42) [inline extensions/skyportal/static/js/components/Filter.jsx:304]: Commented code

**Review by guynir42 — COMMENTED** [review]
Two questions: 
1) should I also go over all the files changed in SkyPortal and Kowalski, or should I assume those changes were already reviewed in other PRs? 
2) What does auto_annotations do in the context of filters?

File: extensions/skyportal/skyportal/handlers/api/kowalski_filter.py [hunk 3]
```diff
@@ -252,14 +241,18 @@ def patch(self, filter_id):
...
         active = data.get("active", None)
         active_fid = data.get("active_fid", None)
-        if (active, active_fid).count(None) != 1:
-            return self.error("One and only one of (active, active_fid) must be set")
+        autosave = data.get("autosave", None)
+        update_annotations = data.get("update_annotations", None)
+        if (active, active_fid, autosave, update_annotations).count(None) < 1:
+            return self.error(
+                "At least one of (active, active_fid, autosave, update_annotations) must be set"
+            )
```
[the remark this answers is not in the document]
Comment (dmitryduev) [reply extensions/skyportal/skyportal/handlers/api/kowalski_filter.py]: Nice catch, this was old logic.

File: extensions/skyportal/static/js/components/Filter.jsx (hunk 2 above)
Comment (dmitryduev) [reply extensions/skyportal/static/js/components/Filter.jsx]: Fixed

**Comment by dmitryduev** [comment]
@guynir42 thanks for the review! You can ignore the K/SP stuff, there are no changes there, it's simple pinning.

File: extensions/skyportal/skyportal/handlers/api/kowalski_filter.py [hunk 4]
```diff
@@ -50,46 +59,42 @@ def get(self, filter_id):
...
-
-            base_url = (
-                f"{self.cfg['app.kowalski.protocol']}://"
-                f"{self.cfg['app.kowalski.host']}:{self.cfg['app.kowalski.port']}"
+        f = (
+            DBSession()
+            .query(Filter)
+            .filter(
+                Filter.id == int(filter_id),
+                Filter.group_id.in_(
```
Comment (acrellin) [inline extensions/skyportal/skyportal/handlers/api/kowalski_filter.py:67]: Should happen in a separate PR, but Fritz extensions need to be updated to use the new access control infrastructure

File: extensions/skyportal/skyportal/handlers/api/kowalski_filter.py [hunk 5]
```diff
@@ -50,46 +59,42 @@ def get(self, filter_id):
...
+        data = response.get("data")
+        # drop monogdb's _id's which are not (default) JSON-serializable
+        if data is not None:
+            data.pop("_id", None)
+        status = response.get("status")
+        message = response.get("message")
+
+        if status == "success":
+            return self.success(data=data)
+        return self.error(message=message)
```
Comment (acrellin) [inline extensions/skyportal/skyportal/handlers/api/kowalski_filter.py:88]:
```suggestion
        if status == "error":
            message = response.get("message")
            return self.error(message=message)
        return self.success(data=data)
```
A bit clearer, and this avoids assigning `message` when it's never accessed in the `success` case

**Review by acrellin — APPROVED** [review]
Looks good!

File: extensions/skyportal/skyportal/handlers/api/kowalski_filter.py (hunk 5 above)
Comment (dmitryduev) [reply extensions/skyportal/skyportal/handlers/api/kowalski_filter.py]: Fixed

File: extensions/skyportal/skyportal/handlers/api/kowalski_filter.py (hunk 4 above)
Comment (dmitryduev) [reply extensions/skyportal/skyportal/handlers/api/kowalski_filter.py:67]: Right, good point

File: extensions/skyportal/skyportal/handlers/api/kowalski_filter.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dmitryduev) [reply extensions/skyportal/skyportal/handlers/api/kowalski_filter.py]: Fixed.

File: extensions/skyportal/skyportal/handlers/api/kowalski_filter.py (hunk 1 above)
Comment (dmitryduev) [reply extensions/skyportal/skyportal/handlers/api/kowalski_filter.py]: Fixed

**Comment by acrellin** [comment]
🚀

**Pull request merged**